src/addon-manager.js

```diff
@@ -377,6 +400,51 @@ class AddonManager extends EventEmitter {
       return Promise.reject(err);
     }
 
+    // Verify the files list in the package.
+    if (!manifest.hasOwnProperty('files') || manifest.files.length === 0) {
+      const err = `files property missing for package ${manifest.name}`;
+      console.error(err);
+      return Promise.reject(err);
+    }
+
+    for (let fname of manifest.files) {
+      fname = path.join(addonPath, fname);
+      if (!fs.existsSync(fname)) {
+        const err = `Package ${manifest.name} missing file: ${fname}`;
+        console.error(err);
+        return Promise.reject(err);
+      }
+    }
+
+    // If a SHA256SUMS file is present, verify it.
+    const sumsFile = path.join(addonPath, 'SHA256SUMS');
```

Comment:
I think that we should provide a comment which shows the format of the SHA256SUMS file. It looks like the file format matches what sha256sum --check would expect, so it would be sufficient to just say that, and perhaps give an example of how to create the SHA256SUM file, something like:
```
cd some-addon-directory
rm -f SHA256SUMS
sha256sum * > SHA256SUMS
```
The rm is needed to ensure that the SHA256SUM file itself doesn't show up in the file. And then you can check using:
```
sha256sum --check SHA256SUMS
```

Reply:
That's exactly what I did in the example and GPIO add-ons, but I can add a comment here.